modules/account-lib/src/coin/eth/transactionBuilder.ts

```diff
@@ -430,16 +430,23 @@ export class TransactionBuilder extends BaseTransactionBuilder {
 
   // set args that are required for all types of eth transactions
   protected buildBase(data: string): TxData {
-    return {
-      _type: ETHTransactionType.LEGACY,
+    const txData = {
```

Comment:
change to

```
  protected buildBase(data: string): TxData {
    const baseParams = {
      gasLimit: this._fee.gasLimit,
      nonce: this._counter,
      data: data,
      chainId: this._common.chainId().toString(),
      value: this._value,
      to: this._contractAddress,
    };

    if (this._fee.eip1559) {
      return {
        ...baseParams,
        _type: ETHTransactionType.EIP1559,
        maxFeePerGas: this._fee.eip1559.maxFeePerGas,
        maxPriorityFeePerGas: this._fee.eip1559.maxPriorityFeePerGas,
      };
    } else {
      return {
        ...baseParams,
        _type: ETHTransactionType.LEGACY,
        gasPrice: this._fee?.gasPrice ?? this._fee.fee,
      };
    }
  }
```

Reply:
Done 
![image](https://user-images.githubusercontent.com/64786258/135102379-565911b7-b87e-4451-9fc6-54d38d32c0b6.png)